### PR TITLE
Bluetooth: Host: Add missing `bt_conn_unref`s

### PR DESCRIPTION
**Pull request by Thalley**
A few places in the host were missing bt_conn_unref in error cases

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

**Review by sjanc — DISMISSED** [review]

File: subsys/bluetooth/host/direction.c
Comment (jori-nordic) [inline subsys/bluetooth/host/direction.c:692]:
```suggestion
		bt_conn_unref(conn);
		
		return;
```
:p

Comment (Thalley) [reply subsys/bluetooth/host/direction.c:692]: Is this a general comment for all the cases, or just this specific one?

Comment (jori-nordic) [reply subsys/bluetooth/host/direction.c:692]: all the cases, mostly tongue-in-cheek since it's a rule you're trying to follow (and make us follow). Im not blocking the PR at all :)

**Review by jori-nordic — DISMISSED** [review]

File: subsys/bluetooth/host/direction.c
Comment (Thalley) [reply subsys/bluetooth/host/direction.c:692]: True :D I can update that no problem

**Review by jori-nordic — DISMISSED** [review]

File: subsys/bluetooth/host/direction.c
Comment (jhedberg) [inline subsys/bluetooth/host/direction.c:873]: Any reason why you use a different style here with no empty line between the unref and the return statement? FWIW, I prefer it like this (no empty line), and it also seems to be the prevalent style elsewhere in this module.

Comment (Thalley) [reply subsys/bluetooth/host/direction.c:873]: The missing newline here was just an oversight :) 

As far as I recall, you were the one who originally reviewed my PRs to add empty lines between expressions and return statements, so I'm surprised to hear that your prefer no empty lines here. 

I can revert it back to not having newlines if that is better, but it's a known issue that we have all these undocumented rules that only occasionally is applied depending on the commit author and reviewer :D

Comment (jhedberg) [reply subsys/bluetooth/host/direction.c:873]: > As far as I recall, you were the one who originally reviewed my PRs to add empty lines between expressions and return statements, so I'm surprised to hear that your prefer no empty lines here.

It's not something I have a super strong opinion on, and it probably depends a bit on the context. E.g. small branches it looks better to me with something more compact. Anyway, no need to get into a debate about it - what trumps style opinions always (even the maintainer's opinion) is IMO consistency, and with an inverse relation to the scope (module more important than subsystem, subsystem more important than project, etc). Since the existing code in this c-file has a certain style, I'd just stick to that.

> I can revert it back to not having newlines if that is better, but it's a known issue that we have all these undocumented rules that only occasionally is applied depending on the commit author and reviewer :D

Yeah I agree, and admit that my guidance can at times seem self contradictory. However, I will always yield to the existing style of a module (although it may need separately pointing out if I don't notice it and request something different).

Comment (jori-nordic) [reply subsys/bluetooth/host/direction.c:873]: Guys I'm sorry I derailed the PR for cosmetic issues, I'm really fine with both. You can revert it to suit johan and I'll approve.

Comment (Thalley) [reply subsys/bluetooth/host/direction.c:869]: Reverted :)

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
